src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayer.java

```diff
@@ -245,22 +264,47 @@ public void addExperience(int amount) {
             // this.experience = (this.experience - 1.0F) * (float)this.xpBarCap();
             // this.addExperienceLevel(1);
         // }
+        postEventAndUpdateExperience(this.experienceTotal + amount);
```

Comment:
Check if there's an event listener with `ShouldFire`, avoids extra events when they're not being used.